include/up-cpp/datamodel/validator/UUri.h

```diff
@@ -49,7 +57,11 @@ std::string_view message(Reason);
 ///     }
 using ValidationResult = std::tuple<bool, std::optional<Reason>>;
 
-/// @brief Checks if UUri is a valid UUri of any format.
+/// @brief Checks if UUri is a valid UUri for use as an attribute in a message
+///
+/// ... but maybe this shouldn't exist. What is allowable is determined by the
+/// message type and is checked using the message validator. I don't think
+/// there is actually a usecase for this anymore.
```

Comment:
Agreed, I like the idea of the message validator checking the UUri's function/type/use explicitly.  Removing this interface seems like a good choice.  I wish there was something between deprecated and removed, like a very-deprecated.

Reply:
Since many uses of up-cpp are likely to have `-Werror` enabled, it's likely that this will have the effect of being "very deprecated" - the compiler will produce an error and refuse to compile calls to deprecated methods.

On a side note, I should revise that comment.